### PR TITLE
The sdk throws foreign JWT exception

### LINKED ISSUE #1018: the sdk throws foreign JWT exception
The sdk contains `com.stormpath.sdk.error.jwt.InvalidJwtException` but in multiple places `io.jsonwebtoken.JwtException` is thrown.

I guess it would be cleaner for the sdk to only throw Stormpath (or general java) exceptions? The fact that you internally are using the io.jsonwebtoken.jwt library should not be visible to the users?

Also you could provide a bit more info to the user on what actually went wrong? I tried to reproduce a case where we see this JwtException from time to time and failed to do so because it could be a lot of things.

https://github.com/stormpath/stormpath-sdk-java/blob/9590dcaa21d7ba2ae46ca05108f87f9cab4e6b79/impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java#L94
https://github.com/stormpath/stormpath-sdk-java/blob/9590dcaa21d7ba2ae46ca05108f87f9cab4e6b79/impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java#L108
https://github.com/stormpath/stormpath-sdk-java/blob/3b5d68792cd60f413427505c7bd7772b95406d77/impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultAccessToken.java#L55

### PR DESCRIPTION
**Pull request by jarias**
Fixes #1018

### CODE REVIEW

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java [hunk 1]
```diff
@@ -29,7 +30,10 @@
 import io.jsonwebtoken.Claims;
 import io.jsonwebtoken.JwtException;
```
Comment (mraible) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java]: This import should be removed.

**Comment by mrioan** [comment]
@lhazlewood not sure about this... 

Is a change in a runtime exception a braking change? I am asking to know whether this should be `2.0.0` or can be `1.2.0`. Please see this as an example: https://github.com/stormpath/stormpath-sdk-java/pull/1021/files#diff-c1b258849dab92454ba040df2550eb84R71. You will see that I am keeping the same `JwtException` in case someone is catching it but they will be lacking the new ones (I am marking it for removal in 2.0.0). Thanks!

**Comment by mrioan** [comment]
This functionality is not critical and will have impact on existent developer's code (i.e. existent code will fail to catch the new exceptions while they are catching them all just fine now). We have therefore decided to postpone this PR until we do 2.0.0

**Comment by mrioan** [comment]
Hey @dogeared Please take [this comment](https://github.com/stormpath/stormpath-sdk-java/pull/1021#issuecomment-254926576) into account. This should not be released until 2.0.0

File: impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java (hunk 1 above)
Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java]: changed `JWT_INVALID_VALUE_ERROR` to `UNSUPPORTED_JWT_ERROR`

Comment (dogeared) [inline impl/src/main/java/com/stormpath/sdk/impl/oauth/DefaultOAuthBearerRequestAuthenticator.java]: @mraible - Would you please give the 👍  and I will merge this into the 2.x branch? Thanks.

**Pull request merged**